diff --git a/bazel/cc_grpc_library.bzl b/bazel/cc_grpc_library.bzl
--- a/bazel/cc_grpc_library.bzl	(revision 78456a56e17c3cbf3b195720934ca3ca460811fa)
+++ b/bazel/cc_grpc_library.bzl	(date 1744843015674)
@@ -26,6 +26,7 @@
         generate_mocks = False,
         use_external = False,
         grpc_only = False,
+        plugin_flags = [],
         **kwargs):
     """Generates C++ grpc classes for services defined in a proto file.

@@ -105,6 +106,7 @@
             plugin = "@com_github_grpc_grpc//src/compiler:grpc_cpp_plugin",
             well_known_protos = well_known_protos,
             generate_mocks = generate_mocks,
+            flags = plugin_flags,
             **kwargs
         )

