diff --git a/src/google/protobuf/BUILD.bazel b/src/google/protobuf/BUILD.bazel
--- a/src/google/protobuf/BUILD.bazel	(revision 5fda5abda3dee5f7a102c85860594bff8d8610bd)
+++ b/src/google/protobuf/BUILD.bazel	(date 1745097172041)
@@ -509,6 +509,7 @@
         "serial_arena.h",
         "thread_safe_arena.h",
         "wire_format_lite.h",
+        "port.h",
     ],
     copts = COPTS + select({
         "//build_defs:config_msvc": [],
diff --git a/python/google/protobuf/__init__.py b/python/google/protobuf/__init__.py
--- a/python/google/protobuf/__init__.py	(revision 5fda5abda3dee5f7a102c85860594bff8d8610bd)
+++ b/python/google/protobuf/__init__.py	(date 1745097184973)
@@ -8,3 +8,9 @@
 # Copyright 2007 Google Inc. All Rights Reserved.
 
 __version__ = '5.28.3'
+
+if __name__ != '__main__':
+  try:
+    __import__('pkg_resources').declare_namespace(__name__)
+  except ImportError:
+    __path__ = __import__('pkgutil').extend_path(__path__, __name__)
