diff --git a/absl/base/config.h b/absl/base/config.h
--- a/absl/base/config.h	(revision 54fac219c4ef0bc379dfffb0b8098725d77ac81b)
+++ b/absl/base/config.h	(date 1744736470656)
@@ -966,7 +966,7 @@
 // https://llvm.org/docs/CompileCudaWithLLVM.html#detecting-clang-vs-nvcc-from-code
 #ifdef ABSL_INTERNAL_HAVE_ARM_NEON
 #error ABSL_INTERNAL_HAVE_ARM_NEON cannot be directly set
-#elif defined(__ARM_NEON) && !defined(__CUDA_ARCH__)
+#elif defined(__ARM_NEON) && !(defined(__NVCC__) && defined(__CUDACC__))
 #define ABSL_INTERNAL_HAVE_ARM_NEON 1
 #endif
 
